src/Moryx.Products.Management/Implementation/Storage/ProductStorage.cs

```diff
@@ -473,20 +472,21 @@ public long SaveType(IProductType modifiedInstance)
         {
             using (var uow = Factory.Create())
             {
-                var entity = SaveProduct(uow, modifiedInstance);
+                var productSaverContext = new ProductPartsSaverContext(uow);
+                var entity = SaveProduct(ref productSaverContext, modifiedInstance);
 
                 uow.SaveChanges();
 
                 return entity.Id;
             }
         }
 
-        private ProductTypeEntity SaveProduct(IUnitOfWork uow, IProductType modifiedInstance)
+        private ProductTypeEntity SaveProduct(ref ProductPartsSaverContext saverContext, IProductType modifiedInstance)
         {
             var strategy = TypeStrategies[modifiedInstance.GetType().Name];
 
             // Get or create entity
-            var repo = uow.GetRepository<IProductTypeEntityRepository>();
+            var repo = saverContext.UnitOfWork.GetRepository<IProductTypeEntityRepository>();
```

Comment:
Maybe we give the context a `GetRepository` method, that forwards the call. What do you think?